systemtest/src/test/java/io/strimzi/systemtest/operators/MultipleClusterOperatorsST.java

```diff
@@ -0,0 +1,247 @@
+        KafkaResource.createAndWaitForReadiness(KafkaResource.kafkaEphemeral(clusterName, 3)
+            .editOrNewMetadata()
+                .addToLabels(FIRST_CO_SELECTOR)
+            .endMetadata()
+            .build());
+
+        LOGGER.info("Removing CR selector from Kafka and increasing number of replicas to 4, new pod should not appear");
+        KafkaResource.replaceKafkaResource(clusterName, kafka -> {
+            kafka.getMetadata().getLabels().clear();
+            kafka.getSpec().getKafka().setReplicas(scaleTo);
+        });
+        KafkaUtils.waitForClusterStability(clusterName);
+
+        Map<String, String> kafkaPods = StatefulSetUtils.ssSnapshot(KafkaResources.kafkaStatefulSetName(clusterName));
+        Map<String, String> zkPods = StatefulSetUtils.ssSnapshot(KafkaResources.zookeeperStatefulSetName(clusterName));
+        Map<String, String> eoPods = DeploymentUtils.depSnapshot(KafkaResources.entityOperatorDeploymentName(clusterName));
+
+        LOGGER.info("Adding CR selector of second CO to Kafka");
+        KafkaResource.replaceKafkaResource(clusterName, kafka -> kafka.getMetadata().setLabels(SECOND_CO_SELECTOR));
+
+        LOGGER.info("Waiting for rolling update because of new label, also Kafka should scale to {}", scaleTo);
+        StatefulSetUtils.waitTillSsHasRolled(KafkaResources.zookeeperStatefulSetName(clusterName), 3, zkPods);
+        StatefulSetUtils.waitTillSsHasRolled(KafkaResources.kafkaStatefulSetName(clusterName), scaleTo, kafkaPods);
+        DeploymentUtils.waitTillDepHasRolled(KafkaResources.entityOperatorDeploymentName(clusterName), 1, eoPods);
+
+        Kafka kafka = KafkaResource.kafkaClient().inNamespace(DEFAULT_NAMESPACE).withName(clusterName).get();
+        assertThat(kafka.getSpec().getKafka().getReplicas(), is(scaleTo));
+        assertThat(StatefulSetUtils.ssSnapshot(KafkaResources.kafkaStatefulSetName(clusterName)).size(), is(scaleTo));
+    }
+
+    @Test
+    void testKafkaCCAndRebalanceWithDifferentCRSelectors() {
+        deployCOInNamespace(FIRST_CO_NAME, DEFAULT_NAMESPACE, FIRST_CO_SELECTOR_ENV, false);
+        deployCOInNamespace(SECOND_CO_NAME, DEFAULT_NAMESPACE, SECOND_CO_SELECTOR_ENV, false);
+
+        LOGGER.info("Deploying Kafka with CruiseControl and CR selector of first CO");
+        KafkaResource.createAndWaitForReadiness(KafkaResource.kafkaWithCruiseControl(clusterName, 3, 3)
+            .editOrNewMetadata()
+                .addToLabels(FIRST_CO_SELECTOR)
+            .endMetadata()
+            .build());
+
+        LOGGER.info("Creating KafkaRebalance with different CR selector than CC");
+        KafkaRebalanceResource.createAndWaitForReadiness(KafkaRebalanceResource.kafkaRebalance(clusterName)
+            .editOrNewMetadata()
+                .addToLabels(SECOND_CO_SELECTOR)
+            .endMetadata()
+            .build());
+
+        KafkaRebalanceUtils.doRebalancingProcess(clusterName);
+    }
```

Comment:
I'm not completely sure this covers the core idea. How do you know from this test that there is only one operator doing something? I think you need to check here that with a wrong label the KafkaRebalance will be ignored as the easiest way to see that. And then fix the label and see it will be used.